src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.LocalFunctions.cs

```diff
@@ -0,0 +1,252 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal partial class DataFlowPass
+    {
+        private readonly SmallDictionary<LocalFunctionSymbol, LocalFuncUsages> _localFuncVarUsages =
+            new SmallDictionary<LocalFunctionSymbol, LocalFuncUsages>();
+
+        private class LocalFuncUsages
+        {
+            public BitVector ReadVars = BitVector.Empty;
+            public BitVector WrittenVars = BitVector.Empty;
+
+            public bool LocalFuncVisited { get; set; } = false;
+        }
+
+        /// <summary>
+        /// At the local function's use site, checks that all variables read
+        /// are assigned and assigns all variables that are definitely assigned
+        /// to be definitely assigned.
+        /// </summary>
+        private void ReplayReadsAndWrites(LocalFunctionSymbol localFunc,
+                                          CSharpSyntaxNode syntax,
+                                          bool writes)
+        {
+            _usedLocalFunctions.Add(localFunc);
+
+            // If we're not inside a local function and the stateChangedAfterUse
+            // variable has been set then we don't want to report anything yet.
```

Comment:
@gafter Turns out the actual mistake was the name of the function: InLocalFunction actually returns true when we're not in a local function. :) I'll be fixing the function implementation.

However, I think this is the correct behavior: when we're not nested inside a local function, any reads that are replayed will not be stored, they will be reported as unassigned. Therefore, it won't help the set of local function reads + writes converge faster, since the set isn't affected by usages outside of local functions, it will also give incorrect results if the set hasn't been settled yet and a diagnostic is prematurely reported.